Assets/Scripts/PlayerMovementController.cs

```diff
@@ -13,11 +13,39 @@ public class PlayerMovementController : MonoBehaviour
 	public float MoveForce = 365f;
 	public float MaxSpeed = 5f;
 	public float JumpForce = 1000f;
+
+	#region Anti Slide Variables and Properties
+
+	[Tooltip("divide player's speed by this amount every time fixed update is being called, only if player is not generating input on horizontal axis, and IsOnIce = false")]
+	[Range(1.1f,50f)]
+	/// <summary>
+	/// divide player's speed by this amount every time fixed update is being called, only if player is not generating input on horizontal axis, and IsOnIce = false, and grounded = true;
+	/// </summary>
+	public float AntiSlideStrength = 10f;
+	[Tooltip("false = apply anti Slide, true = let the player slide")]	
+	/// <summary>
+	/// false = apply anti Slide, false = let the player slide
+	/// </summary>
+	public bool IsOnIce = false;
+
+	private float antiSlide{
+		get{
+			if(AntiSlideStrength <= 1f){
+				Debug.LogWarning("Dividing player's speed by number smaller than 1 would cause the player to gain speed, substituting with 2f instead");
+				return 2f;
```

Comment:
Warning jest OK, ale nie zakazuj ludziom robić czegoś głupiego. It's not a bug, it's a feature.